### PR TITLE
Don't generate clients for example internal APIs

### PR DESCRIPTION
**Pull request by thockin**
Remove the generated code and the code that would generate them.  This kills most of the deltas between generate-groups.sh and generate-internal-groups.sh.  That will be cleaned up further later.

/sig architecture
/kind cleanup

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/code-generator/generate-internal-groups.sh [hunk 1]
```diff
@@ -21,14 +21,13 @@ set -o pipefail
 # generate-internal-groups generates everything for a project with internal types, e.g. an
 # user-provided API server based on k8s.io/apiserver.
 
-if [ "$#" -lt 5 ] || [ "${1}" == "--help" ]; then
+if [ "$#" -lt 4 ] || [ "${1}" == "--help" ]; then
   cat <<EOF
-Usage: $(basename "$0") <generators> <output-package> <internal-apis-package> <extensiona-apis-package> <groups-versions> ...
```
Comment (liggitt) [inline staging/src/k8s.io/code-generator/generate-internal-groups.sh:26]: is this script still actually doing anything with the internal groups? is the intent to delete this script completely by the time you finish carving up https://github.com/kubernetes/kubernetes/pull/114831?

internal and external packages are not always nested (see k8s.io/api and k8s.io/kubernetes/pkg/apis split), so I'm not sure this is still usable by anyone with that split if we collapse the two args to one

File: staging/src/k8s.io/code-generator/generate-internal-groups.sh [hunk 2]
```diff
@@ -57,78 +55,60 @@ shift 5
...
-    ALL_FQ_APIS+=("${EXT_APIS_PKG}/${G}/${V}")
-    EXT_FQ_APIS+=("${EXT_APIS_PKG}/${G}/${V}")
+    FQ_APIS+=("${APIS_PKG}/${G}/${V}")
   done
 done
 
 if [ "${GENS}" = "all" ] || grep -qw "deepcopy" <<<"${GENS}"; then
   echo "Generating deepcopy funcs"
   "${GOPATH}/bin/deepcopy-gen" \
-      --input-dirs "$(codegen::join , "${ALL_FQ_APIS[@]}")" -O zz_generated.deepcopy \
```
Comment (liggitt) [inline staging/src/k8s.io/code-generator/generate-internal-groups.sh:81]: deep-copy generation is still needed on internal types, so it's not clear how the internal package is getting added to the FQ_APIS list

defaulting, conversion, clients, listers, informers are only needed on external types

**Comment by liggitt** [comment]
deletion of the internal clientsets / listers lgtm. a couple questions on the generate-internal-groups.sh change

File: staging/src/k8s.io/code-generator/generate-internal-groups.sh (hunk 1 above)
Comment (thockin) [reply staging/src/k8s.io/code-generator/generate-internal-groups.sh:26]: This script will go away completely, and possibly be relaced by something nicer when go workspaces lands.  AFAICT we have no such users in-tree, but I don't know if we want to maintain this script as an external-facing API.  It *only* works with GOPATH.

File: staging/src/k8s.io/code-generator/generate-internal-groups.sh (hunk 2 above)
Comment (liggitt) [reply staging/src/k8s.io/code-generator/generate-internal-groups.sh:81]: ok, should we stop passing `deepcopy` to generate-internal-groups.sh everywhere then?

File: staging/src/k8s.io/code-generator/generate-internal-groups.sh [hunk 3]
```diff
@@ -57,78 +55,60 @@ shift 5
...
+      --input-dirs "$(codegen::join , "${FQ_APIS[@]}")" -O zz_generated.conversion \
       "$@"
 fi
 
 if [ "${GENS}" = "all" ] || grep -qw "client" <<<"${GENS}"; then
   echo "Generating clientset for ${GROUPS_WITH_VERSIONS} at ${OUTPUT_PKG}/${CLIENTSET_PKG_NAME:-clientset}"
-  if [ -n "${INT_APIS_PKG}" ]; then
-    IFS=" " read -r -a APIS <<< "$(printf '%s/ ' "${INT_FQ_APIS[@]}")"
-    "${GOPATH}/bin/client-gen" \
-        --clientset-name "${CLIENTSET_NAME_INTERNAL:-internalversion}" \
```
Comment (liggitt) [inline staging/src/k8s.io/code-generator/generate-internal-groups.sh:104]: should we stop setting CLIENTSET_NAME_INTERNAL in all callers?

File: staging/src/k8s.io/code-generator/generate-internal-groups.sh (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/code-generator/generate-internal-groups.sh:26]: Dropping the script completely and replacing with something nicer sounds great... as a data point, there's a *lot* of invocations of this [in the wild](https://github.com/search?type=code&q=%2F%5C%2Fgenerate-internal-groups.sh+%5B%22a-z%5C%5C%24%5D%2F+-org%3Akubernetes+-org%3Akubernetes-sigs).

I'd like @sttts' feedback on dropping the ability to run this with internal/external API packages located in different places. If we decide that's ok to drop, should we keep the two args, check they are identical, and error if they aren't? That would preserve existing invocations for folks who aren't relying on internal/external API packages being in different locations.

File: staging/src/k8s.io/code-generator/generate-internal-groups.sh (hunk 2 above)
Comment (thockin) [reply staging/src/k8s.io/code-generator/generate-internal-groups.sh:81]: The script, as it exists, will RE-DO the deepcopy  generation.  I wanted this PR to be just a small self-contained part of the larger PR - maybe easier if I just change the client-gen part and leave the rest for the subsequent commits?

File: staging/src/k8s.io/code-generator/generate-internal-groups.sh (hunk 3 above)
Comment (thockin) [reply staging/src/k8s.io/code-generator/generate-internal-groups.sh:104]: yes, thanks.

**Comment by thockin** [comment]
Simplified this diff, leaving the rest for later, thanks.

**Comment by liggitt** [comment]
> /usr/bin/env: 'python': No such file or directory

uh...

**Comment by liggitt** [comment]
looks like maybe https://github.com/kubernetes/test-infra/pull/28466 broke everything? :)

**Comment by thockin** [comment]
There was a PR yesterday to activate python3 - I justrebased this branch to be sure it's included

**Comment by liggitt** [comment]
commented at https://github.com/kubernetes/test-infra/pull/28466#issuecomment-1399016336 ... the scripts still relying on python2 live in test-infra and are baked into test images

**Comment by liggitt** [comment]
/lgtm
/approve

once presubmits get fixed, should be good to go

**Pull request merged**